### PR TITLE
QATEST-115357 - BO General Sanity - Update/Save client details

### PR DESCRIPTION
**Pull request by Alena-deriv**
Cypress card https://app.clickup.com/t/20696747/CSIT-1097
Test cases https://app.clickup.com/t/20696747/QATEST-115357

Description:
Test is updating client's fist name and last name in BO side, navigates to FE side and checks that changes are correctly reflected there as well.

Screenshot 2024-06-14 at 11 31 06

### CODE REVIEW

File: cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  const characters = 'abcdefghkl'
+  const charactersLength = characters.length
+  for (let i = 0; i < length; i++) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength))
+  }
+  return result
+}
+
+let firstName = userName(5)
+let lastName = userName(7)
```
Comment (bilalmajid-deriv) [inline cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js:14]: I think we have a generate Random name function in helper/utilty can update that to pass an optional parameter to retrieve name with length provided.

Comment (bilalmajid-deriv) [reply cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js:14]: but make sure that it does not effect the other test using that function

File: cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /* Visits BO */
+    cy.visit(BO_URL)
+    cy.findByText('Please login.').click()
+    cy.findByText('Client Management').click()
+    cy.findByPlaceholderText('email@domain.com')
+      .should('exist')
+      .clear()
+      .type(Cypress.env('credentials').test.masterUser.ID)
+    cy.findByRole('button', { name: /View \/ Edit/i }).click()
+    cy.get('.link').eq(1).should('be.visible').click()
```
Comment (bilalmajid-deriv) [inline cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js:33]: if this element is a <a> element tehn can we not try 
cy.findByRole('link','name of link here')

Comment (bilalmajid-deriv) [reply cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js:33]: like below

**Review by bilalmajid-deriv — COMMENTED** [review]
left some recommendations.

File: cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js (hunk 2 above)
Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js:25]: I reused these steps from another BO test, tried to make updates as @bilalmajid-deriv suggested, but test is not able to find the required button and action fails. Left this step as it is

File: cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js (hunk 1 above)
Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/boSanityUpdateSaveClientDetails.cy.js]: @bilalmajid-deriv this is updated to use generateRandomName function

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Pull request merged**